From 5132acd1372e83c3d45c89bd62a75b44aa5438e9 Mon Sep 17 00:00:00 2001
From: Baokun Li <libaokun1@huawei.com>
Date: Wed, 7 Dec 2022 18:37:53 +0800
Subject: [PATCH] dm ioctl: print error when HC and MD do not match
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 5132acd1372e83c3d45c89bd62a75b44aa5438e9
Modified-by-SEL: No


hulk inclusion
category: bugfix
bugzilla: 187327,https://gitee.com/openeuler/kernel/issues/I6111I
CVE: NA

--------------------------------

Print error when hc and md do not match, which is convenient for locating
the cause of the problem

Signed-off-by: Baokun Li <libaokun1@huawei.com>
Reviewed-by: Zhang Yi <yi.zhang@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 drivers/md/dm-ioctl.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/md/dm-ioctl.c b/drivers/md/dm-ioctl.c
index b839705654d4..b012a2748af8 100644
--- a/drivers/md/dm-ioctl.c
+++ b/drivers/md/dm-ioctl.c
@@ -2033,6 +2033,8 @@ int dm_copy_name_and_uuid(struct mapped_device *md, char *name, char *uuid)
 	mutex_lock(&dm_hash_cells_mutex);
 	hc = dm_get_mdptr(md);
 	if (!hc || hc->md != md) {
+		if (hc)
+			DMERR("hash cell and mapped device do not match!");
 		r = -ENXIO;
 		goto out;
 	}
-- 
2.33.0

